### PR TITLE
save alias in history

### LINKED ISSUE #1189: Lookup previous aliases
qTox should provide a way to lookup previous aliases of a particular user. If a lot of users on your friend list change names often, things get messy quickly.

I have no idea what the best way to implement this in the UI is, but my immediate thought is to add an option to the context menu you get when you right-click an item in the friend list. Something like "Show Aliases" that brings up a dialog, perhaps?

### PR DESCRIPTION
**Pull request by PKEv**
fix #1189
 a way to lookup previous aliases of a particular user.

### CODE REVIEW

**Comment by Zer0-One** [comment]
Can you please describe this PR with more than two words? And possibly provide screenshots?

**Comment by tux3** [comment]
Please keep all source code comments in english, I can't merge this as is.

Now the actual code looks ok but storing the nick on each message instead of on each nick change is a bit of a waste of space.

**Comment by PKEv** [comment]
We spent a lot of space but otherwise much more complicated structure

**Comment by tux3** [comment]
It certainly is simple this way.

Ok, so this clearly solves #659, but I don't think it really addresses #1189 

@Zer0-One ?

**Comment by PKEv** [comment]
complete closure of #1189 will allow only the by creating user page (About)

**Comment by tux3** [comment]
Let's start by just closing #659, and we'll see what to do with #1189 then. I think it needs a bit more discussion because I'm not seeing much of a consensus.

File: src/persistence/historykeeper.cpp [hunk 1]
```diff
@@ -244,7 +256,7 @@ void HistoryKeeper::importMessages(const QList<HistoryKeeper::HistMessage> &lst)
     db->exec("BEGIN TRANSACTION;");
     for (const HistMessage &msg : lst)
     {
-        QList<QString> cmds = generateAddChatEntryCmd(msg.chat, msg.message, msg.sender, msg.timestamp, msg.isSent);
+        QList<QString> cmds = generateAddChatEntryCmd(msg.chat, msg.message, msg.sender, msg.timestamp, msg.isSent, QString()); //!!!
```
Comment (agilob) [inline src/persistence/historykeeper.cpp:259]: heh

**Pull request merged**